File: WarhammerAPI/WarhammerAPI/Model/DAL/Entities/Unit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace WarhammerAPI.Model.DAL.Entities
{
    public class Unit
    {
        [Key]
        public int Id { get; set; }
        public UnitType UnitType { get; set; }//hq, elite ....
        public string Faction { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public List<Models> Models { get; set; }
        public List<WargearOptions> WargearOptions { get; set; }
        public UnitNumber UnitNumber { get; set; }
        public bool Character { get; set; }
        public List<Abilities> Abilities { get; set; }
        //public List<Weapon> Wargear { get; set; }
        public List<Degradation> Degradation { get; set; }
        //public int RoleId { get; set; }
        public Role Role { get; set; }
        public List<FractionKeywords> FractionKeywords { get; set; }
        public List<Keywords> Keywords { get; set; }
        public WarlordTrait WarlordTrait { get; set; }
        public bool Relic { get; set; }
    }
}

File: roster-builder-front/src/app/Entities/Unit.ts
import { Keywords } from './Keywords';
import { FractionKeywords } from './FractionKeywords';
import { Degradation } from './Degradation';
import { Abilities } from './Abilities';
import { WargearOptions } from './WargearOptions';
import { Models } from './Models';
import { UnitType } from './UnitType';
import { WarlordTrait } from './WarlordTrait';
import { UnitNumber } from './UnitNumber';
import { Role } from './Role';

export class Unit {
  public id?:number;
  public unitType?: UnitType;
  public faction?: string;
  public name?: string;
  public description?: string;
  public models?: Models[];
  public wargearOptions?: WargearOptions[];
  public unitNumber?: UnitNumber;
  public character?: boolean;
  public abilities?: Abilities[];
  public degradation?: Degradation[];
  public role?: Role;
  public fractionKeywords?: FractionKeywords[];
  public keywords?: Keywords[];
  public warlordTrait?: WarlordTrait;
  public relic?: boolean;
}
